src/Plugin/rest/resource/TriggerPullResource.php

```diff
@@ -92,31 +91,53 @@ public static function create(ContainerInterface $container, array $configuratio
    *
    * @param mixed $data
    *   Data provided from http request.
-   *
-   * @return \Drupal\rest\ModifiedResourceResponse
-   *   The response.
-   *
-   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
    */
   public function post($data) {
-    $status_code = 404;
-
-    // We check for the site uuid and do a pull if found.
     $sent_site_uuid = $data['site_uuid'];
     $remotes = $this->entityTypeManager->getStorage('remote')->loadMultiple();
+    $this->logger->info('Remote contentpool server initiated pull.');
+
+    // Flush response, so that the remote is not blocked by the pulling process.
+    $this->sendResponseOk();
+
     /** @var \Drupal\relaxed\Entity\RemoteInterface $remote */
     foreach ($remotes as $remote) {
       if ($remote_site_uuid = $remote->getThirdPartySetting('contentpool_client', 'remote_site_uuid')) {
         if ($remote_site_uuid == $sent_site_uuid) {
           $this->remotePullManager->doPull($remote, TRUE);
-          $status_code = 200;
         }
       }
     }
 
-    $this->logger->info('Remote contentpool server initiated pull.');
+    // As the Response was already sent, @see sendResponseOk(), the session
+    // can't be accessed anymore and subsequent script execution will trigger a
+    // RuntimeException if we don't exit here.
+    exit();
```

Comment:
This looks good in general, but I'm worried about this exit here, it's not very nice to break the regular request flow as all the terminate events etc. won't flow as usual. Do you think it would be possible to bring this in to the regualr drupal request flow?  \Symfony\Component\HttpFoundation\Response::send() seems to take care of sending the response just fine, so we could just implement terminate  event and add things like ignore_user_abort() in there?

Then you can just return your response as usual in post() + implement an event subscriber to do the rest upon the terminate event when you find your response there.